src/pdf/index.ts

```diff
@@ -49,10 +49,47 @@ export async function generatePdfFromHtml(
 
   const start = Date.now();
   const page = await browser.newPage();
-  await page.setContent(html, { waitUntil: 'load' });
+  await page.setContent(html, { waitUntil: 'networkidle0' });
   await page.emulateMediaType('screen');
+
+  // Extract information about headings and their positions using page.evaluate()
```

Comment:
@yoganandaness maybe it is a good idea to put this part in its own function like for example the case with generateToc down bellow. For example call it `extractHeadingsInfo` and just call it here. I think this as a functionality will be a bit cleaner that way.